common-code/azure_wrappers/azure_container.py

```diff
@@ -0,0 +1,355 @@
+    managed_identity_credential = ManagedIdentityCredential()
+    azure_cli_credential = AzureCliCredential()
+    env_credential = EnvironmentCredential()
+    default_credential = DefaultAzureCredential(
+        exclude_visual_studio_code_credential=True,
+        exclude_interactive_browser_credential=False,
+        interactive_browser_tenant_id=tenant_id,
+        exclude_managed_identity_credential=True,
+    )
+    return ChainedTokenCredential(
+        env_credential,
+        managed_identity_credential,
+        azure_cli_credential,
+        default_credential,
+    )
+
+
+def get_browser_creds():
+    return InteractiveBrowserCredential()
+
+
+def get_container_client_from_url(container_url, credential=None):
+    if credential is None:
+        tenant_id = os.environ.get("TENANT_ID", DEFAULT_TENANT_ID)
+        credential = get_credential(tenant_id)
+    container_client = ContainerClient.from_container_url(
+        container_url=container_url, credential=credential
+    )
+    return container_client
+
+
+def get_container_client(account_url, container_name, credential=None):
+    if credential is None:
+        tenant_id = os.environ.get("TENANT_ID", DEFAULT_TENANT_ID)
+        credential = get_credential(tenant_id)
+    try:
+        container_client = ContainerClient(
+            account_url, container_name, credential=credential
+        )
+    except:
+        print("The specified container does not exist.")
+        sys.exit(1)
+    return container_client
+
+
+def get_blob_service_client(account_url, credential=None):
+    if credential is None:
+        tenant_id = os.environ.get("TENANT_ID", DEFAULT_TENANT_ID)
+        credential = get_credential(tenant_id)
+    blob_service_client = BlobServiceClient(account_url, credential)
+    containers = blob_service_client.list_containers()
+    print(
+        "Successful connection using Azure AD. Available containers are: ",
+        [c.name for c in containers],
+    )
+    return blob_service_client
+
+
+def list_container_files(
+    account_url, container_name, print_names=False, name_starts_with=None
+):
+    """
+    List the files available in a given blob container.
+    Updated to use container client directly rather than from SA client.
+    """
+    tenant_id = os.environ.get("TENANT_ID", DEFAULT_TENANT_ID)
+    credential = get_credential(tenant_id)
+    container_client = ContainerClient(
+        account_url, container_name, credential=credential
+    )
+    blob_list = container_client.list_blobs(name_starts_with=name_starts_with)
+    if print_names:
+        print("The following files are available in", container_name)
+        for blob in blob_list:
+            print("\t" + blob.name)
+    return [blob.name for blob in blob_list]
+
+def get_az_data(
+    account_url,
+    container_name,
+    file_name,
+    return_readable=True,
+    return_stream=False,
+    version_id=None,
+):
+    """
+    Download a csv or excel file from Azure blob storage
```

Comment:
Since we've expanded the function beyond csv and excel files, we should either just say "download a file" or specify supported filetypes.

Reply:
```suggestion
    Download a file from Azure blob storage
```